pkg/quantile/config.go

```diff
@@ -0,0 +1,181 @@
+package quantile
+
+import (
+	"fmt"
+	"math"
+)
+
+const (
+	defaultBinLimit = 4096
+	defaultEps      = 1.0 / 128.0
```

Comment:
Out of curiosity, is there any reason for choosing this particular value of epsilon? This seems a very low relative error guarantee. Even `1.0 / 64.0` should give an almost invisible error in most use cases (namely graphs and histograms) and this would for example give room to use `uint32` counters in bins.

Reply:
It confusingly gets multiplied by 2 in config initialization so it is actually `1/64`
https://github.com/DataDog/datadog-agent/blob/c13a358d9d8fb298e6db6dc1714a2c012ffe2718/pkg/quantile/config.go#L126-L131

The config stuff needs some cleanup, when this is moved to a top level repo I plan on just having these as constants

Reply:
Having them as powers of 2 makes it easier to write a really fast version of `int(math.RoundToEven(c.logGamma(v)))`

Reply:
This test shows a bunch of info about relative/absolute error for a given epsilon.

https://github.com/DataDog/datadog-agent/blob/8d147f85a8a3108a11428459ca9c7f3f35f7c998/pkg/quantile/config_test.go#L151-L186

https://gist.github.com/GeorgeErickson/c3f2819fe13b7c72e6c022514338f252

Reply:
The test doesn't test the `Quantile` function though, does it? I think it would be best to test the quantile function (i.e., the full sketch) so that we catch the errors that are introduced by the interpolation logic, as well as potential systematic errors that may be introduced (I think the current version does introduce a systematic shift of half a bucket).

Multiplying by 2 (i.e., considering `gamma=1+2*eps`) makes sense when we return the middle value of the bucket, because in that case, the relative error is guaranteed to be less than `eps`. Given that we don't return the middle value of the bucket but interpolate instead and therefore, return any possible value in the bucket, then it probably makes more sense to use `gamma=1+eps` indeed, so that the relative error guarantee is still `eps` (although this is the worst case and it should be smaller on average).